src/crimson/osd/pg.cc

```diff
@@ -766,7 +766,10 @@ PG::do_osd_ops(
           }
         }
       } else if (result > 0 && may_write && !rvec) {
-          result = 0;
+        result = 0;
+      } else if (result < 0 && (m->ops.empty() ?
+        0 : m->ops.back().op.flags & CEPH_OSD_OP_FLAG_FAILOK)) {
```

Comment:
Huh, it looks we're handling `CEPH_OSD_OP_FLAG_FAILOK` at two separate layers (`PG` and `OpsExecuter`). What's the motivation behind that?

Reply:
in OpsExecuter, if one osd_op failed but set CEPH_OSD_OP_FLAG_FAILOK, then need continue the next osd_op in m->ops. otherwise break the loop.
in pg, if osd_op set CEPH_OSD_OP_FLAG_FAILOK, should set result = 0 in MOSDOpRelay, but osd_op.rval has the failed error code.

Reply:
@rzarzynski updated, please review again.